src/api.rs

```diff
@@ -98,10 +139,20 @@ pub struct ThreadPool {
 }
 
 impl ThreadPool {
-    pub fn new() -> ThreadPool {
-        ThreadPool {
-            registry: Registry::new()
+    /// Constructs a new thread pool
+    /// If you pass None as arguments, the current number of cores is used
+    /// as determinde via num_cpus::get()
+    /// This function returns Error() if the number of threads is zero
+    /// otherwise it returns Ok()
+    pub fn new(num_threads: Option<usize>) -> Result<ThreadPool,InitResult> {
```

Comment:
It's fine for now, but eventually I would expect this to take a `Configuration` as its arguments.